ponder.config.ts

```diff
@@ -1,29 +1,39 @@
-import { ACTIVE_PLUGIN } from "./src/lib/plugin-helpers";
+import { createConfig } from "ponder";
+import { deepMergeRecursive } from "./src/lib/helpers";
+import { type IntersectionOf, getActivePlugins } from "./src/lib/plugin-helpers";
 import * as baseEthPlugin from "./src/plugins/base.eth/ponder.config";
 import * as ethPlugin from "./src/plugins/eth/ponder.config";
 import * as lineaEthPlugin from "./src/plugins/linea.eth/ponder.config";
 
 const plugins = [baseEthPlugin, ethPlugin, lineaEthPlugin] as const;
 
-// here we export only a single 'plugin's config, by type it as every config
-// this makes all of the mapping types happy at typecheck-time, but only the
-// relevant config is run at runtime
-export default ((): AllConfigs => {
-  const pluginToActivate = plugins.find((p) => p.ownedName === ACTIVE_PLUGIN);
+// The type of the exported default is the intersection of all plugin configs to
+// each plugin can be correctly typechecked
+type AllPluginsConfig = IntersectionOf<(typeof plugins)[number]["config"]>;
 
-  if (!pluginToActivate) {
-    throw new Error(`Unsupported ACTIVE_PLUGIN: ${ACTIVE_PLUGIN}`);
-  }
+const activePlugins = loadActivePlugins();
 
-  pluginToActivate.activate();
+export default (() =>
```

Comment:
can you just return `createConfig(loadActivePlugins()) as AllPluginsConfig`? feels like spreading `contracts: {...activePlugins.contracts}` isn't doing much?

Reply:
Thanks for this suggestion, now it looks way cleaner ;)